pkg/integrations/v2/app_o11y_receiver/handler/handler.go

```diff
@@ -73,33 +73,20 @@ func (ar *AppO11yHandler) HTTPHandler(logger log.Logger) http.Handler {
 		}
 
 		var wg sync.WaitGroup
-		wgDone := make(chan bool)
-		errChan := make(chan error)
 
 		for _, exporter := range ar.exporters {
 			wg.Add(1)
 			go func(exp exporters.AppO11yReceiverExporter) {
 				defer wg.Done()
-				if err = exp.Export(p); err != nil {
+				if err := exp.Export(p); err != nil {
 					level.Error(logger).Log("msg", "exporter error", "exporter", exp.Name(), "error", err.Error())
-					errChan <- err
 				}
 			}(exporter)
 		}
 
-		go func() {
-			wg.Wait()
-			close(wgDone)
-		}()
-
-		select {
-		case <-wgDone:
-			w.WriteHeader(http.StatusAccepted)
-			_, _ = w.Write([]byte("ok"))
-		case err := <-errChan:
-			close(errChan)
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-		}
+		wg.Wait()
+		w.WriteHeader(http.StatusAccepted)
```

Comment:
What should happen in case of an error? Shouldn't we inform the consumer somehow?

Reply:
Couple of things: 
* `StatusAccepted` does not imply that request was successfully proccessed, just that it was a valid request
* Previous implementation intended to report a single error, but would segfault if there was more than one expoter due to concurrency issues. Imho to do proper error reporting we'd need to do some design work first and figure out how best to report multiple errors, what do if some exporters succeeded and some not. I'd wait to see if there's a practical reason for it first

Reply:
If we don't really care about the result then, we can remove the wait group altogether. If the user waits for the full time of the request then we might as well let them know what happened. I do agree that 202 is the proper response here if we want to make this async.

My proposal(s) are:

* Remove blocking wait and return 202
or
* Execute everything synchronously and return the result

I lean towards the first option, as these systems are supposed to be fault tolerant

Reply:
I have thought about this, and would like to keep current approach because:
* Removing blocking makes writing tests harder :frog: how then to block the test until all exporters have finished? probably solvable, but don't wanna spend time on it
* Also by removing blocking frontend won't be able to tell as well if receiver is struggling and limit concurrent requests 
* Again, collecting & errors and figuring out how to communicate them is not easy and not immediately useful, would rather leave this for a separate issue/PR down the line

Reply:
I agree we can skip this for now. As a heads up, I am not questioning one technique over the other, but rather the semantic correctness. The 202 status generally means that the the request has been accepted for processing, which at the point we don't do. We are completing the request normally. To comment on your points

> how then to block the test until all exporters have finished

It depends on the scope of the test. If it is testing the HTTP server, in terms of expected input/output, then testing that the server returns 202 is a form of a test. The async process that completes the request can be treated as a blackbox. I do agree that go makes extremely hard to test this

> Also by removing blocking client won't be able to tell as well if receiver is struggling and limit concurrent requests

Good point. We can solve this by releasing the rate limiting "lease" once the request is completed.

I think we can resolve this convo if that is okay by you

Reply:
when testing http handler, I also wanna make sure that it **does the thing (tm)** , otherwise it's not super useful. Thanks, I'm sure we'll get back to these points soon and improve